modules/miscutil/lib/dbquery.py

```
@@ -25,69 +25,59 @@
 but see the others as well.
 """
 
-__revision__ = "$Id$"
-
 # dbquery clients can import these from here:
 # pylint: disable=W0611
-from MySQLdb import Warning, Error, InterfaceError, DataError, \
-                    DatabaseError, OperationalError, IntegrityError, \
-                    InternalError, NotSupportedError, \
-                    ProgrammingError
+from MySQLdb import (Warning, Error, InterfaceError, DataError,
+                     DatabaseError, OperationalError, IntegrityError,
+                     InternalError, NotSupportedError, ProgrammingError)
+# pylint: enable=W0611
 import gc
 import os
-import string
 import time
 import marshal
 import re
 import atexit
-import os
 
+from contextlib import contextmanager
 from zlib import compress, decompress
+from collections import namedtuple
 from thread import get_ident
-from invenio.config import CFG_ACCESS_CONTROL_LEVEL_SITE, \
-    CFG_MISCUTIL_SQL_USE_SQLALCHEMY, \
-    CFG_MISCUTIL_SQL_RUN_SQL_MANY_LIMIT
+from invenio.config import (CFG_ACCESS_CONTROL_LEVEL_SITE,
+                            CFG_MISCUTIL_SQL_USE_SQLALCHEMY,
+                            CFG_MISCUTIL_SQL_RUN_SQL_MANY_LIMIT,
+                            CFG_PROFILE_DATABASE,
+                            CFG_PROFILE_DATABASE_TO_FILE)
 
 if CFG_MISCUTIL_SQL_USE_SQLALCHEMY:
-    try:
-        import sqlalchemy.pool as pool
-        import MySQLdb as mysqldb
-        mysqldb = pool.manage(mysqldb, use_threadlocal=True)
-        connect = mysqldb.connect
-    except ImportError:
-        CFG_MISCUTIL_SQL_USE_SQLALCHEMY = False
-        from MySQLdb import connect
+    import sqlalchemy.pool as pool
+    import MySQLdb as mysqldb
+    mysqldb = pool.manage(mysqldb, use_threadlocal=True)
+    connect = mysqldb.connect
```

Clearly if you set the FLAG to True, you want the tools and ignoring errors silently is just bad practice.